connectors/sources/github.py

```diff
@@ -787,6 +791,9 @@ async def graphql(self, query, variables=None):
         except BadGraphQLRequest as exception:
             if self.get_rate_limit_encountered(exception.status_code, exception):
                 await self._put_to_sleep(resource_type="graphql")
+            elif exception.status == FORBIDDEN:
+                msg = "Provided GitHub token does not have the necessary permissions to perform the request."
```

Comment:
Let's include the URL in the error message. Perhaps the query too, if that's not risky to have in logs?